### PR TITLE
initial contribution of the serial pushbutton binding

### PR DESCRIPTION
**Pull request by kaikreuzer**
Details about this binding, see the README.md.
This is mainly a very simple implementation that is very helpful for demo setups and which nicely shows the virtues of the new trigger channels.

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.serialbutton/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+## Supported Things
+
+The binding defines a single thing type called `button`.
+
+A button requires the single configuration parameter `port`, which specifies the serial port that should be used. 
+
+The only available channel is a `SYSTEM_RAWBUTTON` channel called `button`, which emits `PRESSED` events (no `RELEASED` events though) whenever data is available on the serial port (which will be read and discarded).
+
+The use case is simple: Connect any push button to pins 2 and 7 or an RS-232 interface as short-cutting those will signal that data is available. Using the default toggle profile, this means that you can use this channel to toggle any Switch item through the button.
```
Comment (sjka) [inline extensions/binding/org.eclipse.smarthome.binding.serialbutton/README.md]: to pins 2 and 7 **of** an RS-232

File: extensions/binding/org.eclipse.smarthome.binding.serialbutton/ESH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="serialbutton" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="button">
+		<label>Push Button</label>
+		<description>A simple push botton</description>
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.serialbutton/ESH-INF/thing/thing-types.xml]: typo: button

File: extensions/binding/org.eclipse.smarthome.binding.serialbutton/ESH-INF/thing/thing-types.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+		<channels>
+			<channel id="button" typeId="system.rawbutton" />
+		</channels>
+
+		<config-description>
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.serialbutton/ESH-INF/thing/thing-types.xml]: re-format the xml

File: extensions/binding/org.eclipse.smarthome.binding.serialbutton/src/main/java/org/eclipse/smarthome/binding/serialbutton/handler/SerialButtonHandler.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+    @Override
+    public void initialize() {
+        String port = (String) getConfig().get(SerialButtonBindingConstants.PARAMETER_CONFIG);
+        if (port == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, "Port must be set!");
+            return;
+        }
+
+        // parse ports and if the default port is found, initialized the reader
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.serialbutton/src/main/java/org/eclipse/smarthome/binding/serialbutton/handler/SerialButtonHandler.java]: type: initialize

**Comment by kaikreuzer** [comment]
Thanks @sjka & @htreu, PR updated accordingly.

**Pull request merged**